snapcraft/libraries.py

```diff
@@ -48,3 +56,40 @@ def _extract_ld_library_paths(ld_conf_file):
                 paths.extend(path_delimiters.split(line))
 
     return paths
+
+
+_libraries = None
+
+
+def _get_system_libs():
+    global _libraries
+    if _libraries:
+        return _libraries
+
+    release = platform.linux_distribution()[1]
+    lib_path = os.path.join(common.get_librariesdir(), release)
+
+    if not os.path.exists(lib_path):
+        logger.warning('No libraries to exclude from this release')
+
+    with open(lib_path) as fn:
+        _libraries = fn.read().split()
```

Comment:
From what I can tell, you could make `_libraries` a `frozenset` and carry on using it as you are, but slightly faster.